docs/Software Engineering Blog/posts/spring-boot-3-1-docker-compose-support-intuitive-consistent-develop-run-tests-containers.md

```diff
@@ -0,0 +1,239 @@
+
+    redis:
+      image: 'redis:7.2'
+      profiles: ["redis", "all"]
+      ports:
+        - '6379:6379'
+  ```
+
+#### Ignore when Building in Image:
+
+[JIB](https://github.com/GoogleContainerTools/jib) is now an out-of-the-box technic of building image for a Spring Boot Application. For now, JIB still include the spring-boot-docker-compose library in the built image (we can track on [this issue](https://github.com/GoogleContainerTools/jib-extensions/issues/158) for new solutions). However, in most cases, we have existed dependency components (e.g. DB service, Redis cluster...) in Test, Staging, or Production environment. So we need to manually exclude it by setting up by jib-layer-filter-extension for [gradle](https://github.com/GoogleContainerTools/jib-extensions/tree/master/first-party/jib-layer-filter-extension-gradle)/[maven](https://github.com/GoogleContainerTools/jib-extensions/tree/master/first-party/jib-layer-filter-extension-maven).
+
+1. add buildscript at the start of root `build.gradle`
+  ```yaml
+  buildscript {
+      dependencies {
+          classpath('com.google.cloud.tools:jib-layer-filter-extension-gradle:0.3.0')
+      }
+  }
+  ... other settings
+  ```
+2. add plugin usage in jib.gradle
+  ```groovy
+  jib {
+    // ... other settings
+    pluginExtensions {
+        pluginExtension {
+            implementation = 'com.google.cloud.tools.jib.gradle.extension.layerfilter.JibLayerFilterExtension'
+            configuration {
+                filters {
+                    filter {
+                        glob = '**/spring-boot-docker-compose-*.jar'
+                    }
+                }
+            }
+        }
+    }
+  }
+  ```
+
+In this way, the image built by JIB will not include the `spring-boot-docker-compose` library and can run like a normal Spring Boot Application. 
+
+### For Testing
+
+#### Activate Docker in Tests:
+
+In default, the `spring-boot-docker-compose` is disabled when running tests. So we need to activate it by ourselves via property: `spring.docker.compose.skip.in-tests`. Besides, the root folder when running a test is the module folder, so we need to provide the file path to the `root/compose.yaml`. In result, we will write a `test/resources/application.yaml` like:
+
+```yaml
+spring:
+  docker:
+    compose:
+      enabled: true
+      file: ../../compose.yml  # my test class is in root/modules/client
+      skip:
+        in-tests: false
+      profiles:
+        active: ${spring.profiles.include}
+```
+
+Then, in our test class, just add the `@SpringBootTest`, so that the docker-compose will trigger by spring boot app.
+
+#### Shared Containers Across Modules:
+
+If your application consists of multiple modules, sharing Docker containers across tests can significantly speed up the
+testing process. Instead of spinning up new containers for each test module, containers can be shared, reducing the startup time and resource usage.
```

Comment:
📝 **[vale]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Google.Passive] In general, use active voice instead of passive voice ('be shared').